app/integration/project-generator/src/main/java/io/syndesis/integration/project/generator/ProjectGenerator.java

```diff
@@ -82,6 +98,94 @@ public InputStream generate(IntegrationDeployment deployment) throws IOException
+            .filter(step -> step.getConnection().isPresent())
+            .forEach(step -> {
+                final String index = step.getMetadata(Step.METADATA_STEP_INDEX).orElseThrow(() -> new IllegalArgumentException("Missing index for step:" + step));
+                final Connection connection = step.getConnection().get();
+                final ConnectorAction action = ConnectorAction.class.cast(step.getAction().get());
+                final ConnectorDescriptor descriptor = action.getDescriptor();
+                final Connector connector = resolveConnector(connection, resourceManager);
+
+                if (connector.getComponentScheme().isPresent() || descriptor.getComponentScheme().isPresent()) {
+                    // Grab the component scheme from the component descriptor or
+                    // from the connector
+                    final String componentScheme = Optionals.first(descriptor.getComponentScheme(), connector.getComponentScheme()).get();
+
+                    Stream.of(connector, connection, step)
+                        .filter(WithConfiguredProperties.class::isInstance)
+                        .map(WithConfiguredProperties.class::cast)
+                        .map(WithConfiguredProperties::getConfiguredProperties)
+                        .flatMap(map -> map.entrySet().stream())
+                        .filter(Predicates.or(connector::isSecret, action::isSecret))
+                        .distinct()
+                        .forEach(
+                            e -> {
+                                String key = String.format("%s-%s.%s", componentScheme, index, e.getKey());
+                                String val = resourceManager.decrypt(e.getValue());
+
+                                properties.put(key, val);
+                            }
+                        );
+                } else {
+                    // The component scheme is defined as camel connector prefix
+                    // for 'old' style connectors.
+                    final String componentScheme = descriptor.getCamelConnectorPrefix();
+
+                    // endpoint secrets
+                    Stream.of(connector, connection, step)
+                        .filter(WithConfiguredProperties.class::isInstance)
+                        .map(WithConfiguredProperties.class::cast)
+                        .map(WithConfiguredProperties::getConfiguredProperties)
+                        .flatMap(map -> map.entrySet().stream())
+                        .filter(Predicates.or(connector::isEndpointProperty, action::isEndpointProperty))
+                        .filter(Predicates.or(connector::isSecret, action::isSecret))
+                        .distinct()
+                        .forEach(
+                            e -> {
+                                String key = String.format("%s-%s.%s", componentScheme, index, e.getKey());
+                                String val = resourceManager.decrypt(e.getValue());
```

Comment:
See comment above about error handling for `decrypt()`.

In addition I wonder whether duplicated code could be reduced (not of high priority).

I'm a bit surprised that there are so many (also non-linear) ways to obtain configured properties. But that is probably due to the fact that we also support 'old style' connectors. It would be really awesome if we could get rid of the old style before GA to get rid of this baggage. wrt. to Scheduler: Wouldn't the timer connector combined with a regular connector (eg. sql connector) also fit the bill ?

And how does the Twitter connector (new style) does the scheduling ?